nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PipelineXml.java

```diff
@@ -0,0 +1,451 @@
+        final String pipelineConfig = context.getProperty(XML_PIPELINE_CONFIG).getValue();
+        final int pipelinePoolSize = context.getProperty(XML_PIPELINE_POOL_SIZE).asInteger();
+        pipelinePool = new ArrayBlockingQueue<>(pipelinePoolSize);
+        for (int i = 0; i < pipelinePoolSize; i++) {
+            final PipelineXmlData pd;
+            if (!StringUtils.isEmpty(pipelineFile)) {
+                pd = new PipelineXmlData(new Input(pipelineFile));
+            } else {
+                pd = new PipelineXmlData(pipelineConfig);
+            }
+            pipelinePool.add(pd);
+        }
+    }
+
+    @OnStopped
+    public void onStopped(final ProcessContext context) {
+        pipelinePool.clear();
+        pipelinePool = null;
+    }
+
+    private void handleInput(PipelineXmlData pd, ProcessContext context, FlowFile original, InputStream stream) throws SaxonApiException {
+        XdmNode inputNode = pd.runtime.parse(new InputSource(stream));
+        pd.pipeline.writeTo(inputPort, inputNode);
+
+        for (final Map.Entry<PropertyDescriptor, String> entry : context.getProperties().entrySet()) {
+            if (!entry.getKey().isDynamic()) {
+                continue;
+            }
+            final QName optName = QName.fromClarkName(entry.getKey().getName());
+            final String value = context.newPropertyValue(entry.getValue()).evaluateAttributeExpressions(original).getValue();
+            pd.pipeline.passOption(optName, new RuntimeValue(value));
+        }
+    }
+
+    private void handleOutput(PipelineXmlData pd, ProcessSession session, FlowFile original, List<Map.Entry<Relationship, FlowFile>> outputs) throws SaxonApiException {
+        int fragmentCount = 0;
+
+        for (final Relationship rel: getRelationships()) {
+            if (rel == REL_PIPELINE_FAILURE || rel == REL_PIPELINE_ORIGINAL) {
+                continue;
+            }
+
+            final ReadablePipe rpipe = pd.pipeline.readFrom(rel.getName());
+            final Serialization serial = pd.pipeline.getSerialization(rel.getName());
+
+            while (rpipe.moreDocuments()) {
+                final XdmNode node = rpipe.read();
+
+                FlowFile outputFF = session.create(original);
+                outputFF = session.write(outputFF, out -> {
+                    final WritableDocument wd = new WritableDocument(pd.runtime, null, serial, out);
+                    wd.write(node);
+                });
+
+                outputFF = session.putAttribute(outputFF, FRAGMENT_INDEX.key(), String.valueOf(fragmentCount++));
+                outputs.add(new SimpleEntry(rel, outputFF));
+            }
+        }
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        PipelineXmlData pd = null;
+        try {
+            pd = pipelinePool.take();
+        } catch (InterruptedException e) {
+            return;
```

Comment:
I suspect instead of returning, this should call Thread.currentThread().interrupt()